### ISSUE TITLE
Problem installing doccano using docker-compose

### ISSUE THREAD

**Issue by rebby123** [doccano/doccano#553]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Microsoft Windows 10 Pro 10.0.18362 
- **Python version**: 

### Describe the problem
When I am trying to use doccano from the docker-compose instructions, the build just hangs and does not install. I have tried several times so I am also removing orphans from previous attempts

### Source code / logs
C:\Users\cs720ro\doccano>docker-compose down --remove-orphans
Removing orphan container "doccano_backend_1"
Removing orphan container "doccano_nginx_1"
Removing orphan container "doccano_frontend_1"
Removing doccano_postgres_1 ... done                                                                                    Removing network doccano_default

C:\Users\cs720ro\doccano> docker-compose -f docker-compose.prod.yml up
Creating network "doccano_default" with the default driver
Creating doccano_frontend_1 ... done                                                                                    Creating doccano_postgres_1 ... done                                                                                    Creating doccano_backend_1  ... done                                                                                    Creating doccano_nginx_1    ... done                                                                                    Attaching to doccano_frontend_1, doccano_postgres_1, doccano_backend_1, doccano_nginx_1
backend_1   | /usr/bin/env: ÔÇÿbash\rÔÇÖ: No such file or directory
postgres_1  | 2020-02-03 14:31:24.616 UTC [1] LOG:  starting PostgreSQL 12.0 on x86_64-pc-linux-musl, compiled by gcc (Alpine 8.3.0) 8.3.0, 64-bit
postgres_1  | 2020-02-03 14:31:24.616 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1  | 2020-02-03 14:31:24.616 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1  | 2020-02-03 14:31:24.635 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1  | 2020-02-03 14:31:24.688 UTC [19] LOG:  database system was shut down at 2020-02-03 14:29:30 UTC
postgres_1  | 2020-02-03 14:31:24.700 UTC [1] LOG:  database system is ready to accept connections
doccano_backend_1 exited with code 127
doccano_frontend_1 exited with code 0
nginx_1     | 2020/02/03 14:31:26 [emerg] 1#1: host not found in upstream "backend" in /etc/nginx/conf.d/nginx.conf:13
nginx_1     | nginx: [emerg] host not found in upstream "backend" in /etc/nginx/conf.d/nginx.conf:13
doccano_nginx_1 exited with code 1

**Comment by icoxfog417** [comment]
Could you please confirm the line ending if you use the PowerShell (https://github.com/puckel/docker-airflow/issues/204#issuecomment-402410556).

If it is the cause, you can avoid it by [git setting](https://git-scm.com/book/it/v2/Customizing-Git-Git-Configuration).

**Comment by icoxfog417** [comment]
related PR #558

**Comment by dentvii** [comment]
i'm still having this issue
Windows 10 Pro
Python

`(t-nlp-env) c:\Users\FVRODRIGUES\GIT>git clone https://github.com/doccano/doccano.git --config core.autocrlf=input
Cloning into 'doccano'...
fatal: unable to access 'https://github.com/doccano/doccano.git/': Failed to connect to github.com port 443: Timed out

(t-nlp-env) c:\Users\FVRODRIGUES\GIT>git clone https://github.com/doccano/doccano.git --config core.autocrlf=input
Cloning into 'doccano'...
remote: Enumerating objects: 18, done.
remote: Counting objects: 100% (18/18), done.
remote: Compressing objects: 100% (17/17), done.
remote: Total 8967 (delta 6), reused 2 (delta 1), pack-reused 8949
Receiving objects: 100% (8967/8967), 43.90 MiB | 6.89 MiB/s, done.
Resolving deltas: 100% (6269/6269), done.

(t-nlp-env) c:\Users\FVRODRIGUES\GIT>cd doccano

(t-nlp-env) c:\Users\FVRODRIGUES\GIT\doccano>docker-compose -f docker-compose.prod.yml up
Recreating doccano_frontend_1 ... done                                                                                  Starting doccano_postgres_1   ... done                                                                                  Recreating doccano_backend_1  ... done                                                                                  Recreating doccano_nginx_1    ... done                                                                                  Attaching to doccano_postgres_1, doccano_frontend_1, doccano_backend_1, doccano_nginx_1
backend_1   | /usr/bin/env: ‘bash\r’: No such file or directory
postgres_1  | 2020-02-18 11:46:45.338 UTC [1] LOG:  starting PostgreSQL 12.0 on x86_64-pc-linux-musl, compiled by gcc (Alpine 8.3.0) 8.3.0, 64-bit
postgres_1  | 2020-02-18 11:46:45.338 UTC [1] LOG:  listening on IPv4 address "0.0.0.0", port 5432
postgres_1  | 2020-02-18 11:46:45.338 UTC [1] LOG:  listening on IPv6 address "::", port 5432
postgres_1  | 2020-02-18 11:46:45.364 UTC [1] LOG:  listening on Unix socket "/var/run/postgresql/.s.PGSQL.5432"
postgres_1  | 2020-02-18 11:46:45.407 UTC [18] LOG:  database system was shut down at 2020-02-17 07:07:37 UTC
postgres_1  | 2020-02-18 11:46:45.416 UTC [1] LOG:  database system is ready to accept connections
doccano_backend_1 exited with code 127
doccano_frontend_1 exited with code 0
nginx_1     | 2020/02/18 11:46:46 [emerg] 1#1: host not found in upstream "backend" in /etc/nginx/conf.d/nginx.conf:13
nginx_1     | nginx: [emerg] host not found in upstream "backend" in /etc/nginx/conf.d/nginx.conf:13
doccano_nginx_1 exited with code 1
`

Conda lista

`# Name                    Version                   Build  Channel
appdirs                   1.4.3                    pypi_0    pypi
attrs                     19.1.0                   pypi_0    pypi
autopep8                  1.4.4                      py_0
backcall                  0.1.0                    pypi_0    pypi
bleach                    3.1.0                    pypi_0    pypi
blis                      0.2.4                    pypi_0    pypi
boto3                     1.12.0                   pypi_0    pypi
botocore                  1.15.0                   pypi_0    pypi
ca-certificates           2020.1.1                      0
catalogue                 1.0.0                      py_0    conda-forge
certifi                   2019.11.28               py37_0
cffi                      1.13.2           py37hb32ad35_0    conda-forge
chardet                   3.0.4                 py37_1003    conda-forge
click                     7.0                      pypi_0    pypi
colorama                  0.4.3                    pypi_0    pypi
cryptography              2.8              py37hb32ad35_1    conda-forge
cudatoolkit               10.1.243             h74a9793_0
cupy-cuda101              7.2.0                    pypi_0    pypi
cymem                     2.0.3            py37h6538335_0    conda-forge
cython-blis               0.4.1            py37hfa6e2cd_0    conda-forge
decorator                 4.4.1                    pypi_0    pypi
defusedxml                0.6.0                    pypi_0    pypi
docutils                  0.15.2                   pypi_0    pypi
entrypoints               0.3                      pypi_0    pypi
fastrlock                 0.4                      pypi_0    pypi
flask                     1.1.1                    pypi_0    pypi
flask-api                 2.0                      pypi_0    pypi
ftfy                      5.6                      pypi_0    pypi
idna                      2.8                      pypi_0    pypi
importlib_metadata        1.5.0                    py37_0    conda-forge
intel-openmp              2020.0                      166
ipykernel                 5.1.4                    pypi_0    pypi
ipython                   7.12.0                   pypi_0    pypi
ipython-genutils          0.2.0                    pypi_0    pypi
itsdangerous              1.1.0                    pypi_0    pypi
jedi                      0.16.0                   pypi_0    pypi
jinja2                    2.10.3                   pypi_0    pypi
jmespath                  0.9.4                    pypi_0    pypi
joblib                    0.14.1                   pypi_0    pypi
jsonschema                3.2.0                    py37_0    conda-forge
jupyter-client            5.3.4                    pypi_0    pypi
jupyter-core              4.6.2                    pypi_0    pypi
label-studio              0.4.6.post2              pypi_0    pypi
label-studio-converter    0.0.10                   pypi_0    pypi
libblas                   3.8.0                    14_mkl    conda-forge
libcblas                  3.8.0                    14_mkl    conda-forge
liblapack                 3.8.0                    14_mkl    conda-forge
lxml                      4.2.5                    pypi_0    pypi
markupsafe                1.1.1                    pypi_0    pypi
mistune                   0.8.4                    pypi_0    pypi
mixpanel                  4.4.0                    pypi_0    pypi
mkl                       2019.4                      245
murmurhash                1.0.0            py37h6538335_0    conda-forge
nbconvert                 5.6.1                    pypi_0    pypi
nbformat                  5.0.4                    pypi_0    pypi
ninja                     1.9.0            py37h74a9793_0
notebook                  6.0.3                    pypi_0    pypi
numpy                     1.17.4                   pypi_0    pypi
openssl                   1.1.1d               he774522_4
orjson                    2.3.0                    pypi_0    pypi
pandas                    1.0.1                    pypi_0    pypi
pandocfilters             1.4.2                    pypi_0    pypi
parso                     0.6.1                    pypi_0    pypi
pickleshare               0.7.5                    pypi_0    pypi
pillow                    6.2.0                    pypi_0    pypi
pip                       20.0.2                   py37_1
plac                      0.9.6                    pypi_0    pypi
preshed                   2.0.1                    pypi_0    pypi
prometheus-client         0.7.1                    pypi_0    pypi
prompt-toolkit            3.0.3                    pypi_0    pypi
pt-core-news-sm           2.2.5                    pypi_0    pypi
pycodestyle               2.5.0                    py37_0
pycparser                 2.19                     py37_1    conda-forge
pygments                  2.5.2                    pypi_0    pypi
pyheartex                 0.0.2                     dev_0    <develop>
pyopenssl                 19.1.0                   py37_0    conda-forge
pyrsistent                0.15.7                   pypi_0    pypi
pysocks                   1.7.1                    py37_0    conda-forge
python                    3.7.6                h60c2a47_2
python-dateutil           2.8.1                    pypi_0    pypi
python-json-logger        0.1.11                   pypi_0    pypi
pytorch                   1.4.0           py3.7_cuda101_cudnn7_0    pytorch
pytz                      2019.3                   pypi_0    pypi
pywin32                   227                      pypi_0    pypi
pywinpty                  0.5.7                    pypi_0    pypi
pyyaml                    5.3                      pypi_0    pypi
pyzmq                     18.1.1                   pypi_0    pypi
rarfile                   3.1                      pypi_0    pypi
redis                     3.4.1                    pypi_0    pypi
regex                     2020.1.8                 pypi_0    pypi
requests                  2.22.0                   py37_1    conda-forge
rq                        1.0                      pypi_0    pypi
s3transfer                0.3.3                    pypi_0    pypi
sacremoses                0.0.38                   pypi_0    pypi
scikit-learn              0.21.3                   pypi_0    pypi
scipy                     1.4.1                    pypi_0    pypi
send2trash                1.5.0                    pypi_0    pypi
sentencepiece             0.1.85                   pypi_0    pypi
setuptools                45.2.0                   py37_0
six                       1.12.0                   pypi_0    pypi
spacy                     2.2.3            py37he980bc4_0    conda-forge
spacy-transformers        0.5.1                    pypi_0    pypi
sqlite                    3.31.1               he774522_0
srsly                     0.2.0                    pypi_0    pypi
terminado                 0.8.3                    pypi_0    pypi
testpath                  0.4.4                    pypi_0    pypi
thinc                     7.3.1                    pypi_0    pypi
torchcontrib              0.0.2                    pypi_0    pypi
tornado                   6.0.3                    pypi_0    pypi
tqdm                      4.40.2                   pypi_0    pypi
traitlets                 4.3.3                    pypi_0    pypi
transformers              2.0.0                    pypi_0    pypi
urllib3                   1.25.7                   py37_0    conda-forge
vc                        14.1                 h0510ff6_4
vs2015_runtime            14.16.27012          hf0eaf9b_1
wasabi                    0.4.2                    pypi_0    pypi
wcwidth                   0.1.8                    pypi_0    pypi
webencodings              0.5.1                    pypi_0    pypi
werkzeug                  0.16.0                   pypi_0    pypi
wheel                     0.34.2                   py37_0
win_inet_pton             1.1.0                    py37_0    conda-forge
wincertstore              0.2                      py37_0
xmljson                   0.2.0                    pypi_0    pypi
zipp                      2.2.0                      py_0    conda-forge`

![image](https://user-images.githubusercontent.com/87960/74733564-54eb4e80-522b-11ea-8ffc-a5d080a77778.png)

**Comment by icoxfog417** [comment]
@dentvii Please read `Note for Windows developers`. It is caused by line breaks in the bash file.

https://github.com/doccano/doccano#docker-compose

**Comment by dentvii** [comment]
Sorry,  didn’t understand as I have used the git config aas shown in the image
`git clone https://github.com/doccano/doccano.git --config core.autocrlf=input `
Yet the results are the same. Is there anything else I am missing?

**Comment by icoxfog417** [comment]
The bash files in your cloned repository will contain `crlf` line breaks. Please replace these to `lf`.

**Comment by dentvii** [comment]
Manually changed all CRLF to LF from all SH files within doccano folder, yet yielded the same results

![image](https://user-images.githubusercontent.com/87960/74846571-6e65c680-530f-11ea-934c-a3d91c32e520.png)

**Comment by icoxfog417** [comment]
Would you please delete images that built from docker-compose not only containers? Your images will already be built with `crlf` bash files.

**Comment by dentvii** [comment]
@icoxfog417 made it!

For Windows Users:

1. Update GIT ([Download new version here](https://git-scm.com/download/win))
2. Open terminal
3. Type  `git --version`
4. Check if the version matches
5. Create a new folder
6. Use the following command `git clone https://github.com/doccano/doccano.git --config core.autocrlf=input`
7. After the download, open Notepad++ ([downloadable here](https://notepad-plus-plus.org/downloads/)) on split screen
8. Within Notepad++, on the menu click on View, Show Symbol, then Enable Show End of Line
9. ![image](https://user-images.githubusercontent.com/87960/75084710-abdd7600-5500-11ea-8931-705ad408987a.png)
10. Now within the folder of the Git type on the search bar *.sh
11. Select all of these files and drop them inside Notepad++
12. If you see this, the core.autocrlf=input failed
13. ![image](https://user-images.githubusercontent.com/87960/75084751-0d054980-5501-11ea-8e2a-5aec3e343274.png)
14. If it failed, go back to the folder search and select all files (yml, py, sh) and drop them in NotePad++
15. Now use the shortcut CTRL+H and type the following, with attention to the options
16. ![image](https://user-images.githubusercontent.com/87960/75084784-5e153d80-5501-11ea-8597-a66f131b2844.png)
17. Save all files 
18. If you finished this or if you didn't have any problem at all with CR LF,  now go inside the folder created by the git command
19. Now use the folllowing command `docker-compose -f docker-compose.prod.yml up`

It will take a long time, but it should now build.
You will not seem it working 0.0.0.0, but rather localhost:80/

IF you have tried to build before without the error, you will have to clear the docker images.
To do so, stop all containers (I'll assume you don't have any competing dockers)

1. Use the command `docker container ls` to see all the containers 
2. Now use the command `docker container stop ` followed by the copied and paste of all the rows in the column CONTAINER_ID
3. Clear the containers with `docker system prune`
4. List all images with `docker images`
5. Now use the comand `docker rmi --force ` followed by the copy and paste of all the rows in IMAGE_ID (if they don't delete, you may try again as the order is important)
6. It is useful to restart the Docker on Windows (use the GUI) or if you can't find it just restart windows itself.
7. Now go back to the guide above

**Comment by kut** [comment]
thanks folks. btw i had this error too and cloning using:
`git clone https://github.com/doccano/doccano.git --config core.autocrlf=false`
...fixed the issue and allowed docker compose to work
